src/MichiTokenizeRequestor.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "tokenbound/src/AccountV3Upgradable.sol";
+
+import "./interfaces/IMichiWalletReceiptNFT.sol";
+
+/**
+ * @title MichiTokenizeRequestor
+ *     @dev Implementation of a request contract. Users can create requests to tokenize
+ *     points earned in ERC-6551 accounts (TBAs). Once the request is made, the tokenized
+ *     points will be minted and the Michi Wallet NFT will be transferred to the MichiTBALocker.
+ */
+contract MichiTokenizeRequestor is Ownable {
+    using SafeERC20 for IERC20;
+    using Counters for Counters.Counter;
+
+    /// @notice counter to create a unique id for each request
+    Counters.Counter private _requestIds;
+
+    struct Request {
+        address requestor;
+        address michiWalletAddress;
+        uint256 requestId;
+    }
+
+    /// @notice address of the MichiTBALocker
+    address public michiTBALocker;
+
+    /// @notice instance of the receipt NFT
+    IMichiWalletReceiptNFT public michiWalletReceiptNFT;
+
+    /// @notice enter id to retrieve request
+    mapping(uint256 => Request) public idToRequest;
+
+    /// @notice mapping of approved Michi Wallet collections
+    mapping(address => bool) public approvedCollections;
```

Comment:
might want to rename to isCollectionApproved for clarity. Or you could use a set instead of a mapping.

Reply:
renamed